airbyte-config/persistence/src/main/java/io/airbyte/config/persistence/ConfigRepository.java

```diff
@@ -627,35 +636,84 @@ public void updateConnectionState(final UUID connectionId, final State state) th
     return actorCatalogs;
   }
 
-  public void writeCatalog(final AirbyteCatalog catalog,
-                           final UUID sourceId,
-                           final String configurationHash,
-                           final String connectorVersion)
-      throws JsonValidationException, IOException {
+  private Map<UUID, AirbyteCatalog> findCatalogByHash(final String catalogHash, final DSLContext context) {
+    final Result<Record2<UUID, JSONB>> records = context.select(ACTOR_CATALOG.ID, ACTOR_CATALOG.CATALOG)
+        .from(ACTOR_CATALOG)
+        .where(ACTOR_CATALOG.CATALOG_HASH.eq(catalogHash)).fetch();
+
+    final Map<UUID, AirbyteCatalog> result = new HashMap<>();
+    for (final Record record : records) {
+      final AirbyteCatalog catalog = Jsons.deserialize(
+          record.get(ACTOR_CATALOG.CATALOG).toString(), AirbyteCatalog.class);
+      result.put(record.get(ACTOR_CATALOG.ID), catalog);
+    }
+    return result;
+  }
+
+  private UUID getOrInsertActorCatalog(final AirbyteCatalog airbyteCatalog,
+                                       final DSLContext context) {
+    final OffsetDateTime timestamp = OffsetDateTime.now();
```

Comment:
nit: I think we usually use `Instant` and `Instant.now()` for cases like this. Is there a particular reason why you chose `OffsetDateTime` here? It's not a big deal either way - both should work fine

Reply:
`DatabaseConfigPersistence` seems to rely only `OffsetDateTime.now()` only and does not use `Instant` at all. So I used the same object to be consistent with other database access patterns.

I just tried using `Instant.now()` and I'm getting the following error:
> no suitable method found for set(TableField<ActorCatalogRecord,OffsetDateTime>,Instant)

We can try to normalize the usage of DateTime library but I think it should be handled as a separate ticket/set of PRs

Reply:
Got it, I did not realize that `DatabaseConfigPersistence` used this as well. No worries, then lets go with this